### PR TITLE
tplg: add nocodec topology for icelake

### PR DESCRIPTION
**Pull request by RanderWang**
Test it on ICL

Signed-off-by: Rander Wang <rander.wang@linux.intel.com>

### CODE REVIEW

File: topology/sof-icl-nocodec.m4 [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Icelake DSP configuration
+include(`platform/intel/icl.m4')
+
+#
+# Define the pipelines
+#
+# PCM0 ----> volume -----> volume ---->  SSP0
```
Comment (plbossart) [inline topology/sof-icl-nocodec.m4]: why do you need two volumes?

File: topology/sof-icl-nocodec.m4 [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	PIPELINE_SINK_2, 2, s24le,
+	48, 1000, 0, 0)
+
+# PCM Low Latency
+PCM_DUPLEX_ADD(Passthrough, 3, 0, 0, PIPELINE_PCM_1, PIPELINE_PCM_2)
+
+#
+# BE configurations - overrides config in ACPI if present
+#
+DAI_CONFIG(SSP, 0, 0, NoCodec-0,
```
Comment (plbossart) [inline topology/sof-icl-nocodec.m4]: this looks like all other test topologies, why do you need something specific for ICL?

File: topology/platform/intel/icl.m4 [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	MEMCAPS(MEM_CAP_RAM, MEM_CAP_DMA, MEM_CAP_CACHE, MEM_CAP_HP))
+define(`PLATFORM_HOST_MEM_CAP',
+	MEMCAPS(MEM_CAP_RAM, MEM_CAP_DMA, MEM_CAP_CACHE, MEM_CAP_HP))
+define(`PLATFORM_PASS_MEM_CAP',
+	MEMCAPS(MEM_CAP_RAM, MEM_CAP_DMA, MEM_CAP_CACHE, MEM_CAP_HP))
+define(`PLATFORM_COMP_MEM_CAP', MEMCAPS(MEM_CAP_RAM, MEM_CAP_CACHE))
+
+# Low Latency PCM Configuration
+# Low Latency PCM Configuration
+W_VENDORTUPLES(pipe_ll_schedule_plat_tokens, sof_sched_tokens, LIST(`		', `SOF_TKN_SCHED_MIPS	"50000"'))
```
Comment (plbossart) [inline topology/platform/intel/icl.m4]: fix the alignment?

File: topology/sof-icl-nocodec.m4 [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	   SSP_CONFIG(DSP_B, SSP_CLOCK(mclk, 38400000, codec_mclk_in),
```
Comment (plbossart) [inline topology/sof-icl-nocodec.m4]: can you check if the MCLK really needs to be that high? Or are we confusing clock source and clock output?

Comment (RanderWang) [reply topology/sof-icl-nocodec.m4]: yes, MCLK on ICL is 38.4MHZ confirmed by AUDIO PEG team

File: topology/platform/intel/icl.m4 (hunk 3 above)
Comment (RanderWang) [reply topology/platform/intel/icl.m4]: yes, I will

File: topology/sof-icl-nocodec.m4 (hunk 1 above)
Comment (RanderWang) [reply topology/sof-icl-nocodec.m4]: typo, I will fix it

File: topology/sof-icl-nocodec.m4 (hunk 2 above)
Comment (RanderWang) [reply topology/sof-icl-nocodec.m4]: some settings are different on different platform. For example, there are 6 SSP ports available on APL, but only 2 on CNL. And MCLK on APL is 19.2MHZ, but on CNL is 24MHZ and on ICL 38.4MHZ. So it is trouble for others to validate on each platform. So I made this for others to have a easy life

File: topology/platform/intel/icl.m4 [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+define(`PLATFORM_DAI_MEM_CAP',
+	MEMCAPS(MEM_CAP_RAM, MEM_CAP_DMA, MEM_CAP_CACHE, MEM_CAP_HP))
+define(`PLATFORM_HOST_MEM_CAP',
+	MEMCAPS(MEM_CAP_RAM, MEM_CAP_DMA, MEM_CAP_CACHE, MEM_CAP_HP))
+define(`PLATFORM_PASS_MEM_CAP',
+	MEMCAPS(MEM_CAP_RAM, MEM_CAP_DMA, MEM_CAP_CACHE, MEM_CAP_HP))
+define(`PLATFORM_COMP_MEM_CAP', MEMCAPS(MEM_CAP_RAM, MEM_CAP_CACHE))
+
+# Low Latency PCM Configuration
+W_VENDORTUPLES(pipe_ll_schedule_plat_tokens, sof_sched_tokens, LIST(`		', `SOF_TKN_SCHED_MIPS	"50000"'))
```
Comment (plbossart) [inline topology/platform/intel/icl.m4]: can you fix the long lines?

Comment (RanderWang) [reply topology/platform/intel/icl.m4]: yes, I will fix it

**Comment by RanderWang** [comment]
Fix log lines issue

**Pull request merged**